From 035c67006ec04cce72c573652c77f6a4b262e54b Mon Sep 17 00:00:00 2001
From: Adam Ehlers Nyholm Thomsen <adament@adament.net>
Date: Sat, 19 Oct 2024 07:02:33 +0200
Subject: [PATCH] Install osqp_demo and osqp_tester

This makes them relocatable and allows running them in the test step or
after a user has installed the package, according to h-vetinari they
would be included in the artifact anyway for running tests locally.
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6992004..04ba244b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -366,6 +366,7 @@ if (NOT PYTHON AND NOT MATLAB AND NOT R_LANG AND NOT EMBEDDED)
     # Create demo executable (linked to static library)
     add_executable (osqp_demo ${PROJECT_SOURCE_DIR}/examples/osqp_demo.c)
     target_link_libraries (osqp_demo osqpstatic)
+	install(TARGETS osqp_demo)
 
 endif (NOT PYTHON AND NOT MATLAB AND NOT R_LANG AND NOT EMBEDDED)
 
@@ -465,6 +466,7 @@ if (UNITTESTS)
                 CXX_EXTENSIONS NO
             )
     target_link_libraries (osqp_tester osqpstatic ${CMAKE_DL_LIBS})
+	install(TARGETS osqp_tester)
 
     # Add custom memory target
     add_executable(osqp_tester_custom_memory
